src/packageurl/__init__.py

```diff
@@ -228,13 +233,13 @@ def normalize_subpath(subpath, encode=True):  # NOQA
     return subpath or None
 
 
-def normalize(type, namespace, name, version, qualifiers, subpath, encode=True):  # NOQA
+def normalize(type, namespace, name, version, qualifiers, subpath, encode=True, name_has_slash=False):  # NOQA
```

Comment:
Do you need this `name_has_slash=False` arg here and elsewhere instead of just checking if there is a `/` in the name?

Reply:
I believe so. It is valid to pass in a `<namespace>/<name>` combination, which is indistinguishable from a `<name with a / in it>`. The only way I see to specify which thing we're passing in is with an optional argument. I'm open to better ideas though.